drivers/serial/uart_mcux_iuart.c

```diff
@@ -0,0 +1,307 @@
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+	u32_t mask = kUART_TxEmptyEnable;
+
+	return (UART_GetEnabledInterrupts(config->base) & mask)
+		&& mcux_iuart_irq_tx_complete(dev);
+}
+
+static void mcux_iuart_irq_rx_enable(struct device *dev)
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+	u32_t mask = kUART_RxDataReadyEnable;
+
+	UART_EnableInterrupts(config->base, mask);
+}
+
+static void mcux_iuart_irq_rx_disable(struct device *dev)
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+	u32_t mask = kUART_RxDataReadyEnable;
+
+	UART_DisableInterrupts(config->base, mask);
+}
+
+static int mcux_iuart_irq_rx_full(struct device *dev)
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+
+	return (UART_GetStatusFlag(config->base, kUART_RxDataReadyFlag)) != 0U;
+}
+
+static int mcux_iuart_irq_rx_ready(struct device *dev)
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+	u32_t mask = kUART_RxDataReadyFlag;
+
+	return (UART_GetEnabledInterrupts(config->base) & mask)
+		&& mcux_iuart_irq_rx_full(dev);
+}
+
+static void mcux_iuart_irq_err_enable(struct device *dev)
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+	u32_t mask = kUART_RxOverrunEnable | kUART_ParityErrorEnable |
+			kUART_FrameErrorEnable;
+
+	UART_EnableInterrupts(config->base, mask);
+}
+
+static void mcux_iuart_irq_err_disable(struct device *dev)
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+	u32_t mask = kUART_RxOverrunEnable | kUART_ParityErrorEnable |
+			kUART_FrameErrorEnable;
+
+	UART_DisableInterrupts(config->base, mask);
+}
+
+static int mcux_iuart_irq_is_pending(struct device *dev)
+{
+	return mcux_iuart_irq_tx_ready(dev) || mcux_iuart_irq_rx_ready(dev);
+}
+
+static int mcux_iuart_irq_update(struct device *dev)
+{
+	return 1;
+}
+
+static void mcux_iuart_irq_callback_set(struct device *dev,
+				       uart_irq_callback_user_data_t cb,
+				       void *cb_data)
+{
+	struct mcux_iuart_data *data = dev->driver_data;
+
+	data->callback = cb;
+	data->cb_data = cb_data;
+}
+
+static void mcux_iuart_isr(void *arg)
+{
+	struct device *dev = arg;
+	struct mcux_iuart_data *data = dev->driver_data;
+
+	if (data->callback) {
+		data->callback(data->cb_data);
+	}
+}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+static int mcux_iuart_init(struct device *dev)
+{
+	const struct mcux_iuart_config *config = dev->config->config_info;
+	uart_config_t uart_config;
+	struct device *clock_dev;
+	u32_t clock_freq;
+
+	clock_dev = device_get_binding(config->clock_name);
+	if (clock_dev == NULL) {
+		return -EINVAL;
+	}
+
+	if (clock_control_get_rate(clock_dev, config->clock_subsys,
+				   &clock_freq)) {
+		return -EINVAL;
+	}
+
+	UART_GetDefaultConfig(&uart_config);
+	uart_config.enableTx = true;
+	uart_config.enableRx = true;
+	uart_config.baudRate_Bps = config->baud_rate;
+
+	UART_Init(config->base, &uart_config, clock_freq);
+
```

Comment:
The irq_config_func is never called and should be called in init.
`#ifdef CONFIG_UART_INTERRUPT_DRIVEN                                                                                                                                                                            
         config->irq_config_func(dev);                                                                                                                                                                           #endif`